Tests/src/PublicAPITests/HotspotSequenceTests.cpp

```diff
@@ -689,4 +692,400 @@ CSP_PUBLIC_TEST(CSPEngine, HotspotSequenceTests, GenerateSequenceKeyTest)
+
+	bool ComponentAdded = false;
+
+	CreatedObject->SetUpdateCallback(
+		[&ComponentAdded, ObjectName](csp::multiplayer::SpaceEntity* Entity,
+									  csp::multiplayer::SpaceEntityUpdateFlags Flags,
+									  csp::common::Array<csp::multiplayer::ComponentUpdateInfo>& UpdateInfo)
+		{
+			if (Entity->GetName() == ObjectName)
+			{
+				for (size_t i = 0; i < UpdateInfo.Size(); ++i)
+				{
+					if (UpdateInfo[i].UpdateType == csp::multiplayer::ComponentUpdateType::Add)
+					{
+						ComponentAdded = true;
+					}
+				}
+			}
+		});
+
+	// Create hotspot component
+	auto* HotspotComponent
+		= static_cast<csp::multiplayer::HotspotSpaceComponent*>(CreatedObject->AddComponent(csp::multiplayer::ComponentType::Hotspot));
+
+	CreatedObject->QueueUpdate();
+	WaitForCallbackWithUpdate(ComponentAdded, EntitySystem);
+
+	EXPECT_TRUE(ComponentAdded);
+
+	// Create Hotspot groups
+	csp::common::String TestGroupName1 = "CSP-UNITTEST-SEQUENCE-MAG1";
+	csp::common::String TestGroupName2 = "CSP-UNITTEST-SEQUENCE-MAG2";
+	csp::common::String TestItemName   = "AnotherItem";
+
+	{
+		// Create 2 groups that contains the component
+
+		// Create one with only a single item to test deletion functionality
+		csp::systems::HotspotGroup HotspotGroup1;
+		CreateHotspotgroup(HotspotSystem, TestGroupName1, {HotspotComponent->GetUniqueComponentId()}, HotspotGroup1);
+
+		// Create one with an additional item to test update functionality
+		csp::systems::HotspotGroup HotspotGroup2;
+		CreateHotspotgroup(HotspotSystem, TestGroupName2, {HotspotComponent->GetUniqueComponentId(), TestItemName}, HotspotGroup2);
+
+		// Ensure the 2 groups are created correctly
+		csp::common::Array<csp::systems::HotspotGroup> FoundGroups;
+		GetHotspotGroups(HotspotSystem, {TestGroupName1, TestGroupName2}, FoundGroups);
+
+		EXPECT_EQ(FoundGroups.Size(), 2);
+	}
+
+	// Remove component
+	{
+		uint16_t ComponentId  = HotspotComponent->GetId();
+		bool ComponentRemoved = false;
+
+		CreatedObject->SetUpdateCallback(
+			[&ComponentRemoved, ObjectName, ComponentId](csp::multiplayer::SpaceEntity* Entity,
+														 csp::multiplayer::SpaceEntityUpdateFlags Flags,
+														 csp::common::Array<csp::multiplayer::ComponentUpdateInfo>& UpdateInfo)
+			{
+				if (Entity->GetName() == ObjectName)
+				{
+					for (size_t i = 0; i < UpdateInfo.Size(); ++i)
+					{
+						if (UpdateInfo[i].UpdateType == csp::multiplayer::ComponentUpdateType::Delete)
+						{
+							ComponentRemoved = true;
+						}
+					}
+				}
+			});
+
+		CreatedObject->RemoveComponent(HotspotComponent->GetId());
+
+		CreatedObject->QueueUpdate();
+		WaitForCallbackWithUpdate(ComponentRemoved, EntitySystem);
+
+		EXPECT_TRUE(ComponentRemoved);
+
+		// We currently need to wait some extra time, because our callback doesnt rely on the internal sequence calls finishing.
+		// We shouldn't need to do this after completing OF-1191 as we can do this in the new deletion endpoint callback
+		std::this_thread::sleep_for(3000ms);
```

Comment:
Some risk here that the sleep time is only assumed to be long enough, but not guaranteed.
We could instead await for the sequence delete multiplayer callback that gets fired to avoid this.

Reply:
I'm going to update OnLocalDelete to use a callback to fix this. Only downside is deletion will take longher if it contains hotspots

Reply:
Actually, using the multiplayer callback is a good shout, as waiting for the callbacks was becoming unmanagable.